orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -106,6 +106,14 @@ public long getTimeout() {
     return TimeUnit.MINUTES.toMillis(5);
   }
 
+  public long getDynamicTimeout(StageExecution stage) {
+    if (stage.getContext().containsKey("timeout")) {
```

Comment:
it would be nice to reuse (if possible) `public static final String STAGE_TIMEOUT_OVERRIDE_KEY = "stageTimeoutMs";` which already read in the `StageExecution`  - code snipped below from `StageExecutionImpl`

```

public Optional<Long> getTimeout() {
  Object timeout = getContext().get(STAGE_TIMEOUT_OVERRIDE_KEY);
  if (timeout instanceof Integer) {
    return Optional.of((Integer) timeout).map(Integer::longValue);
  } else if (timeout instanceof Long) {
    return Optional.of((Long) timeout);
  } else if (timeout instanceof Double) {
    return Optional.of((Double) timeout).map(Double::longValue);
  }
  return Optional.empty();
}
```

Reply:
I tested using this value, but it is used in other places too and I don't think it should be update from the ui. As per Slack chat, I changed 'timeout' -> 'rollbackTimeout' for better understanding.